homeassistant/components/telegram_bot/__init__.py

```diff
@@ -380,31 +388,62 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     if not domain_config:
         return False
 
+    # create config entries from YAML
+    for config_import in domain_config:
+        trusted_networks: list[IPv4Network] = config_import.get(
+            CONF_TRUSTED_NETWORKS, []
+        )
+        trusted_networks_str: list[str] = (
+            [str(trusted_network) for trusted_network in trusted_networks]
+            if trusted_networks
+            else []
+        )
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={CONF_SOURCE: SOURCE_IMPORT},
+                data={
+                    CONF_PLATFORM: config_import.get(CONF_PLATFORM),
+                    CONF_API_KEY: config_import.get(CONF_API_KEY),
+                    CONF_ALLOWED_CHAT_IDS: config_import.get(CONF_ALLOWED_CHAT_IDS),
+                    ATTR_PARSER: config_import.get(ATTR_PARSER),
+                    CONF_PROXY_URL: config_import.get(CONF_PROXY_URL),
+                    CONF_URL: config_import.get(CONF_URL),
+                    CONF_TRUSTED_NETWORKS: trusted_networks_str,
+                },
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: TelegramBotConfigEntry) -> bool:
+    """Create the Telegram bot from config entry."""
     platforms = await async_get_loaded_integration(hass, DOMAIN).async_get_platforms(
```

Comment:
I don't think we need to use `async_get_loaded_integration`. We can just import the "platform" modules at the top of this module and store them in a constant dict with the platform type as key and look up the correct module to use here.